### ISSUE TITLE
[MJARSIGNER-30] IT failures with Jdk 8 (EA)

### ISSUE THREAD

**Issue by jira-importer** [apache/maven-jarsigner-plugin#102]
**[Hervé Boutemy](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=hboutemy)** opened **[MJARSIGNER-30](https://issues.apache.org/jira/browse/MJARSIGNER-30?redirect=false)** and commented

see https://builds.apache.org/view/M-R/view/Maven/job/maven-plugins-ITs-m3.1.x-with-maven-plugin-jdk-1.8/modules

```
[INFO] --- maven-invoker-plugin:1.7:verify (integration-test) @ maven-jarsigner-plugin ---
[INFO] -------------------------------------------------
[INFO] Build Summary:
[INFO]   Passed: 7, Failed: 1, Errors: 0, Skipped: 0
[INFO] -------------------------------------------------
[ERROR] The following builds failed:
[ERROR] *  verify-fail/pom.xml
[INFO] -------------------------------------------------
```

---

**Affects:** 1.2

**Issue Links:**
- [MNG-5551](https://issues.apache.org/jira/browse/MNG-5551) check that everything compile and run fine with Java 8

**Comment by jira-importer** [comment]
**[Robert Scholte](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=rfscholte)** commented

```
[main] [DEBUG] Processing E:\java-workspace\apache-maven-plugins\maven-jarsigner-plugin\target\it\verify-fail\tampered.jar
[main] [DEBUG] cmd.exe /X /C ""C:\Program Files\Java\jdk1.7.0_45\jre\..\bin\jarsigner.exe" -verify -keystore E:\java-workspace\apache-maven-plugins\maven-jarsigner-plugin/src/it/keystore E:\java-workspace\apache-maven-plugins\maven-jarsigner-plugin\target\it\verify-fail\tampered.jar"
[main] [DEBUG] Executing: cmd.exe /X /C ""C:\Program Files\Java\jdk1.7.0_45\jre\..\bin\jarsigner.exe" -verify -keystore E:\java-workspace\apache-maven-plugins\maven-jarsigner-plugin/src/it/keystore E:\java-workspace\apache-maven-plugins\maven-jarsigner-plugin\target\it\verify-fail\tampered.jar"
[Thread-2] [DEBUG] jarsigner: java.lang.SecurityException: Invalid signature file digest for Manifest main attributes
[main] [INFO] ------------------------------------------------------------------------
[main] [INFO] BUILD FAILURE
[main] [INFO] ------------------------------------------------------------------------
```

```
[main] [DEBUG] Processing E:\java-workspace\apache-maven-plugins\maven-jarsigner-plugin\target\it\verify-fail\tampered.jar
[main] [DEBUG] cmd.exe /X /C ""c:\Program Files\Java\jdk1.8.0\jre\..\bin\jarsigner.exe" -verify -keystore E:\java-workspace\apache-maven-plugins\maven-jarsigner-plugin/src/it/keystore E:\java-workspace\apache-maven-plugins\maven-jarsigner-plugin\target\it\verify-fail\tampered.jar"
[main] [DEBUG] Executing: cmd.exe /X /C ""c:\Program Files\Java\jdk1.8.0\jre\..\bin\jarsigner.exe" -verify -keystore E:\java-workspace\apache-maven-plugins\maven-jarsigner-plugin/src/it/keystore E:\java-workspace\apache-maven-plugins\maven-jarsigner-plugin\target\it\verify-fail\tampered.jar"
[Thread-2] [DEBUG] jar is unsigned. (signatures missing or not parsable)
[main] [INFO] 1 archive(s) processed
[main] [INFO] ------------------------------------------------------------------------
[main] [INFO] BUILD SUCCESS
[main] [INFO] ------------------------------------------------------------------------
```

So the behavior of the jarsigner executable has changed.

**Comment by jira-importer** [comment]
**[Tony Chemit](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=tchemit)** commented

With jdk <1.8, when can not verify a jar then exit code is 1 (this is normal); while with jdk 1.8, result code is 0.

IMHO, this is a bug from jdk 1.8: when a command fails, you can't have a result code to 0.

We nned to open/track this **regression**.

**Comment by jira-importer** [comment]
**[Robert Scholte](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=rfscholte)** commented

JDK 8 Build b126 Early Access comes with a fix for [JDK-8031572](https://bugs.openjdk.java.net/browse/JDK-8031572)
Now the issue can't be reproduced anymore.

**Comment by jira-importer** [comment]
**[Karl Heinz Marbaise](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=khmarbaise)** commented

Does this mean we can close the issue? as fixed ?
